azkaban-common/src/main/java/azkaban/executor/container/ContainerizedDispatchManager.java

```diff
@@ -475,10 +475,28 @@ public void pauseFlow(ExecutableFlow exFlow, String userId) throws ExecutorManag
     throw new UnsupportedOperationException("Unsupported Method");
   }
 
-  //TODO: BDP-3642 Add a way to call Flow container APIs using apiGateway
   @Override
   public void retryFailures(ExecutableFlow exFlow, String userId) throws ExecutorManagerException {
-    throw new UnsupportedOperationException("Unsupported Method");
+    synchronized (exFlow) {
+      final Map<Integer, Pair<ExecutionReference, ExecutableFlow>> unfinishedFlows = this.executorLoader
+          .fetchUnfinishedFlows();
+      if (unfinishedFlows.containsKey(exFlow.getExecutionId())) {
+        final Pair<ExecutionReference, ExecutableFlow> pair = unfinishedFlows
```

Comment:
Can you please add log line in this method? It will help later on.

Reply:
done